http4k-core/src/test/kotlin/org/http4k/lens/BiDiLensSpecTest.kt

```diff
@@ -112,6 +113,13 @@ class BiDiLensSpecTest {
     @Test
     fun yearMonth() = checkContract(spec.yearMonth(), YearMonth.of(2000, 2), "2000-02", "", "invalid", "o", "o2000-02", "o2000-022000-02")
 
+    @Test
+    fun enum() {
+        fun BiDiLensSpec<String, String>.enum() = map(StringBiDiMappings.enum<Method>())
```

Comment:
Please note, that I declare the `BiDiLensSpec.enum()` method here. The reasons to not have it defined publicly in `lensSpec.kt` like so
```kotlin
inline fun <IN : Any, reified T: Enum<T>> BiDiLensSpec<IN, String>.enum() = map(StringBiDiMappings.enum<T>())
```
are that
1) It would have no usages, because both `Path` and `Query` use `StringBiDiMappings.enum` directly.
2) unfortunately Kotlin is not able to infer the `IN` type when calling the method on the `Query` object. Calling the `enum` extension then becomes impractical
```kotlin
// Request must be specified again
Query.enum<Request, MyEnum>().required("...")
```

I still however wanted to add a test using the `checkContract` helper that's why the test ended up here. In fact it tests functionality from `BiDiMappings.kt` and not `lensSpec.kt` but that's true for most of the tests in this file.

I'm not sure if I took the right decision to play along the general structure of this project.